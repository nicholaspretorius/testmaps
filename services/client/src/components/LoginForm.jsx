import React from "react";

const LoginForm = () => {
  return (
    <div>
      <h1 className="title is-1">Login</h1>
      <hr />
      <br />
    </div>
  );
};

export default LoginForm;
