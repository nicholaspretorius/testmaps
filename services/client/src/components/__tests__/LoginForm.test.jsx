import React from "react";
import { cleanup, render } from "@testing-library/react";

import LoginForm from "./../LoginForm";

afterEach(cleanup);

describe("renders", () => {
  const props = {
    onHandleLoginFormSubmit: () => {
      return true;
    },
    isAuthenticated: () => {
      return false;
    }
  };

  it("properly", () => {
    const { getByText } = renderWithRouter(<LoginForm {...props} />);
    expect(getByText("Login")).toHaveClass("title");
  });

  it("with default props", () => {
    const { getByLabelText, getByText } = renderWithRouter(<LoginForm {...props} />);

    const emailInput = getByLabelText("Email");
    expect(emailInput).toHaveAttribute("type", "email");
    expect(emailInput).not.toHaveValue();

    const passwordInput = getByLabelText("Password");
    expect(passwordInput).toHaveAttribute("type", "password");
    expect(passwordInput).not.toHaveValue();

    const buttonInput = getByText("Submit");
    expect(buttonInput).toHaveValue("Submit");
  });

  it("a snapshot", () => {
    const { asFragment } = renderWithRouter(<LoginForm {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
