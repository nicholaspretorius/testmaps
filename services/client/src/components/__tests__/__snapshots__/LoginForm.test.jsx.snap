// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="title is-1"
    >
      Login
    </h1>
    <hr />
    <br />
    <form>
      <div
        class="field"
      >
        <label
          class="label"
          for="input-email"
        >
          Email
        </label>
        <input
          class="input"
          id="input-email"
          name="email"
          placeholder="Enter your email address"
          type="email"
          value=""
        />
      </div>
      <div
        class="field"
      >
        <label
          class="label"
          for="input-password"
        >
          Password
        </label>
        <input
          class="input"
          id="input-password"
          name="password"
          placeholder="Enter your password"
          type="password"
          value=""
        />
      </div>
      <input
        class="button is-primary"
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</DocumentFragment>
`;
